ipywidgets/static/widgets/js/widget.js

```diff
@@ -642,12 +642,17 @@ define(["nbextensions/widgets/widgets/js/manager",
              */
             switch(value) {
                 case null: // python None
-                    this.el.style.display = '';
+                    if (this._old_display) {
+                        this.el.style.display = this._old_display;
+                    }
                     this.el.style.visibility = 'hidden'; break;
                 case false:
+                    this._old_display = this.el.style.display;
                     this.el.style.display = 'none'; break;
                 case true:
-                    this.el.style.display = '';
+                    if (this._old_display) {
+                        this.el.style.display = this._old_display;
+                    }
```

Comment:
If display was inherited to start with, and you hide it, would _old_display be set to 'block'?